src/mesh/mesh.cpp

```diff
@@ -911,6 +914,22 @@ void Mesh::ApplyUserWorkBeforeOutput(Mesh *mesh, ParameterInput *pin,
   }
 }
 
+//
+//----------------------------------------------------------------------------------------
+// \!fn void Mesh::ApplyUserWorkBeforeRestartOutput
+// \brief Apply Mesh and Meshblock versions of UserWorkBeforeRestartOutput
+void Mesh::ApplyUserWorkBeforeRestartOutput(Mesh *mesh, ParameterInput *pin,
+                                            SimTime const &time) {
+
+  // call Mesh version
+  mesh->UserMeshWorkBeforeRestartOutput(mesh, pin, time);
+
+  // call MeshBlock version
+  for (auto &pmb : block_list) {
+    pmb->UserWorkBeforeRestartOutput(pmb.get(), pin);
+  }
+}
+
```

Comment:
Should this be in `Mesh`? Or would it be better to put it in `Driver`? I lean towards `Driver`.

Reply:
I put this right below `Mesh::ApplyUserWorkBeforeOutput`, if we want to move this we should probably move both.

Reply:
Ah ok. Leave it for now then... My preference is to move all of this stuff into Driver and `StateDescriptor` rather than looping it through Mesh. But that's a bigger refactor maybe.

Reply:
Yeah I'd rather defer that and build consensus first